pkg/proxy/proxy.go

```diff
@@ -373,3 +389,48 @@ func replaceTokenInWebsocketRequest(req *http.Request, newToken string) {
 	}
 	req.Header.Set(ph, strings.Join(protocols, ","))
 }
+
+func validateWorkspaceRequest(requestedWorkspace, requestedNamespace string, workspaces []toolchainv1alpha1.Workspace) error {
+	// check workspace access
+	isHomeWSRequested := requestedWorkspace == ""
+
+	allowedWorkspace := -1
+	for i, w := range workspaces {
+		if w.Name == requestedWorkspace || (isHomeWSRequested && w.Status.Type == "home") {
+			allowedWorkspace = i
+			break
+		}
+	}
```

Comment:
can there be more workspaces of `.Status.Type: home` in the `workspaces` list ? 
In that case it will provide access to the first one it finds if I'm not wrong.

Reply:
There should be only 1 home workspace for a user. Only the space created for the user when they first sign up should be their home workspace and remain their home workspace. We have some plans to ensure there can only be 1 home workspace. See https://github.com/codeready-toolchain/registration-service/blob/4b359be48a41f69fb832c82b333b630210cc5d6e/pkg/proxy/handlers/spacelister.go#L154

Reply:
Great! Thanks for the additional details!